From: Guido Urdaneta <guidou@webrtc.org>
Date: Thu, 24 Jul 2025 11:01:29 +0200
Subject: (cherry-pick-branch-heads/7258) Use FieldTrialsView::IsEnabled for
 DTLS 1.3

No behavior changes.

(cherry picked from commit 5ff715d5666106e01d27205c1775d1e2d07ea254)

Bug: webrtc:383141571, chromium:433885045, chromium:434133034
Change-Id: Ice5f3e5cbd245ddea407248a6f29c61c646e6a72
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/401740
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Commit-Queue: Guido Urdaneta <guidou@webrtc.org>
Cr-Original-Commit-Position: refs/heads/main@{#45206}
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/402200
Cr-Commit-Position: refs/branch-heads/7258@{#3}
Cr-Branched-From: 74fa937f86ed8432c07676f7a1ce0e5e2812b3d5-refs/heads/main@{#44974}
---
 rtc_base/openssl_stream_adapter.cc | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/rtc_base/openssl_stream_adapter.cc b/rtc_base/openssl_stream_adapter.cc
index 7d7466b1cc..604a9465c7 100644
--- a/rtc_base/openssl_stream_adapter.cc
+++ b/rtc_base/openssl_stream_adapter.cc
@@ -144,13 +144,15 @@ int GetForceDtls13(const FieldTrialsView* field_trials) {
     return kForceDtls13Off;
   }
 #ifdef DTLS1_3_VERSION
-  auto mode = field_trials->Lookup("WebRTC-ForceDtls13");
-  RTC_LOG(LS_WARNING) << "WebRTC-ForceDtls13: " << mode;
-  if (mode == "Enabled") {
+  if (field_trials->IsEnabled("WebRTC-ForceDtls13")) {
+    RTC_LOG(LS_WARNING) << "WebRTC-ForceDtls13 Enabled";
     return kForceDtls13Enabled;
-  } else if (mode == "Only") {
+  }
+  if (field_trials->Lookup("WebRTC-ForceDtls13") == "Only") {
+    RTC_LOG(LS_WARNING) << "WebRTC-ForceDtls13 Only";
     return kForceDtls13Only;
   }
+  RTC_LOG(LS_WARNING) << "WebRTC-ForceDtls13 Disabled";
 #endif
   return kForceDtls13Off;
 }
